middlewares/studentAssessmentAccess.js

```diff
@@ -0,0 +1,168 @@
+//var ERR = require('async-stacktrace');
+//var _ = require('lodash');
+//var sha256 = require('crypto-js/sha256');
+const express = require('express');
+var router = express.Router();
+
+var logger = require('../lib/logger');
+var csrf = require('../lib/csrf');
+var config = require('../lib/config');
+
+var timeout = 24; // hours
+
+router.get('/', function(req, res, next) {
+
+// Future feature
+//    // SEB: check BrowserExamKey
+//    if ('x-safeexambrowser-requesthash' in req.headers) {
+//        checkBrowserExamKeys(res, req.headers['x-safeexambrowser-requesthash']);
+//    }
+
+    // SEB: check user-agent
+    if ('user-agent' in req.headers) {
+        checkUserAgent(res, req.headers['user-agent']);
+    }
+
+// Future feature
+//    // Course Instance view (only show allowed exam)
+//    if (!('assessment' in res.locals)) {
+//        //console.log('In Course Instance Page');
+//        //res.locals.authz_data.mode = 'SEB';
+//        //console.log(res.locals);
+//    }
+
+    // SafeExamBrowser protect the assesment
+    if ('authz_result' in res.locals &&
+        res.locals.authz_result.mode == 'SEB') {
+
+        // If the assessment is complete, use this middleware to show the logout page
+        if ('assessment_instance' in res.locals && res.locals.assessment_instance.open == false) {
+            return badSEB(req, res);
+        }
+
+        // If any of our auth checks didn't pass, fail (send to download)
+        if (res.locals.authz_data.mode != 'SEB') {
+            return badSEB(req, res);
+        }
+    }
+
+
+    // Password protect the assessment
```

Comment:
Can we separate out the password stuff into a separate middleware? I think there isn't much connecting passsword and SEB, right?

Reply:
They are independent but work in similar ways and would be called in the same places. Since they're both just a few small functions, I included them in the single middleware.